### PR TITLE
FIX: make asset registry thread safe

### PR DESCRIPTION
**Pull request by tacaswell**
This also has a few other clean ups to account for h5py
and Python moving under us.

### CODE REVIEW

File: databroker/assets/sqlite.py [hunk 1]
```diff
@@ -137,6 +192,62 @@ def reconnect(self):
...
+                # queue.
+                continue
+            try:
+                finished_event, name, args, kwargs, ret = item
+            except ValueError:
+                print("did not get the right number of values in {item}".
+                      format(item=item))
+                continue
+            try:
+                print(f'handling {name}')
```
Comment (danielballan) [inline databroker/assets/sqlite.py]: Stray print.

Comment (danielballan) [reply databroker/assets/sqlite.py]: I pushed a commit to fix this since I have checked it out anyway. :-)

Comment (tacaswell) [reply databroker/assets/sqlite.py]: Thanks.

File: databroker/assets/base_registry.py [hunk 2]
```diff
@@ -472,12 +472,14 @@ def register_resource(self, spec, root, rpath, rkwargs,
...
-                                         root=root,
-                                         path_semantics=path_semantics,
-                                         **rs_kwarg)['uid']
+        return self._api.insert_resource(
+            col, spec, rpath,
+            rkwargs,
+            self.known_spec,
+            root=root,
+            path_semantics=path_semantics,
+            duplicate_exc=self.DuplicateKeyError,
```
Comment (danielballan) [inline databroker/assets/base_registry.py:481]: Good catch.

**Comment by danielballan** [comment]
I checked the CI failures, and they are all related to server startup timeouts, safe to ignore.

File: databroker/assets/base_registry.py (hunk 2 above)
Comment (tacaswell) [reply databroker/assets/base_registry.py:481]: This was failing on Python master (as it was trying to do `except None` and at some point throwing / catching things that do not derive from BaseException will be forbidden).

**Pull request merged**